src/cmds/scripts/pbs_snapshot

```diff
@@ -35,20 +35,62 @@
 # "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
 # trademark licensing policies.
 
-pbs_conf=${PBS_CONF_FILE:-/etc/pbs.conf}
-if [ -r "${pbs_conf}" ]; then
-	. ${pbs_conf}
+ptllibpath=
+ptlbinpath=
+
+# Try rpm install paths
+if [ -f "/etc/debian_version" ]; then
+        ptl_prefix_lib=$(dpkg -L pbspro-ptl 2>/dev/null | grep -m 1 lib$ 2>/dev/null)
 else
-	echo "***" >&2
-	echo "*** No PBS_CONF_FILE found, exiting" >&2
-	echo "***" >&2
-	exit 1
+        ptl_prefix_lib=$(rpm -ql pbspro-ptl 2>/dev/null | grep -m 1 lib$ 2>/dev/null)
+fi
+
+# Try system paths
+if [ ! -d  ${ptl_prefix_lib} ]; then
+    ptl_prefix_lib=/usr/local/lib/python3.[[:digit:]]/site-packages
+    if [ ! -d  ${ptl_prefix_lib} ]; then
+        ptl_prefix_lib=/usr/lib/python3.[[:digit:]]/site-packages
+    fi
 fi
 
-export PYTHONPATH=${PBS_EXEC}/unsupported/fw:${PYTHONPATH}
+if [ "x${ptl_prefix_lib}" != "x" ]; then
+        python_dir=$(/bin/ls -1 ${ptl_prefix_lib})
+        prefix=$(dirname ${ptl_prefix_lib})
+
+        ptllibpath=${prefix}/lib/${python_dir}/site-packages
+        ptlbinpath=${prefix}/bin
+else
+        conf="${PBS_CONF_FILE:-/etc/pbs.conf}"
+        if [ -r "${conf}" ]; then
+                # we only need PBS_EXEC from pbs.conf
+                __PBS_EXEC=$( grep '^[[:space:]]*PBS_EXEC=' "$conf" | tail -1 | sed 's/^[[:space:]]*PBS_EXEC=\([^[:space:]]*\)[[:space:]]*/\1/' )
+                if [ "X${__PBS_EXEC}" != "X" ]; then
+                        # Define PATH and PYTHONPATH for the users
+                        PTL_PREFIX=$(dirname ${__PBS_EXEC})/ptl
```

Comment:
I don't understand this part. Are we looking for 'ptl' in PBS_EXEC path? What I understood is we will come to this else block only if PTL rpm is not installed, right? From pbspro.spec, `%define ptl_prefix %{pbs_prefix}/../ptl`

Reply:
Ignore this comment. I understood it.